src/controller/CommissioningWindowOpener.h

```diff
@@ -107,6 +163,38 @@ class CommissioningWindowOpener
                                        Callback::Callback<OnOpenCommissioningWindow> * callback, SetupPayload & payload,
                                        bool readVIDPIDAttributes = false);
 
+    /**
+     * @brief
+     *   Try to look up the device attached to our controller with the given
+     *   node id and ask it to re-enter commissioning mode with a PASE verifier
+     *   derived from the given information and the given discriminator. The
+     *   device will exit commissioning mode after a successful commissioning,
+     *   or after the given `timeout` time.
+     *
+     * @param[in] params        The parameters required to open an enhanced commissioning window
+     *                          with the provided or generated passcode.
+     * @param[out] payload      The setup payload, not including the VID/PID bits,
+     *                          even if those were asked for, that is generated
+     *                          based on the passed-in information.  The payload
+     *                          provided to the callback function, unlike this
+     *                          out parameter, will include the VID/PID bits if
+     *                          readVIDPIDAttributes is true.
+     */
+    CHIP_ERROR OpenCommissioningWindow(const CommissioningWindowPasscodeParams & params, SetupPayload & payload);
```

Comment:
Why have a passcode version? The verifier version beloew is sufficient.

Reply:
This is to move off the previous function with growing number of args (for the passcode version) as requested in https://github.com/project-chip/connectedhomeip/pull/34209#discussion_r1667046245. The existing API is there only to reduce the external SDK client churn (could potentially mark it as _deprecated_ in some form, or note for consumers to move off that?)